### DISCUSSION TITLE
ESP32-S3, ST7789, 16-bit Interface Help

### DISCUSSION THREAD

**Discussion by RDJunc** [moononournation/Arduino_GFX#540] [Q&A]
Hi.  I am working on a project that uses a Newhaven 2.4" 240-320 TFT display with a touch screen.  It uses a ST7789 display driver.  The board is controlled with an ESP32-S3.  Since I have lots of GPIO, I have set it up to use the full 16-bit parallel interface.

Information on using any parallel interface is pretty limited.  It is pretty much focused on SPI.  The information about this library says that it supports a 16-bit parallel interface but I cannot find any more about it.

Can someone please point me in the right direction when it comes to setting up the ESP32 to use this display and with the 16-bit interface?

Thanks very much.

**Comment by mrmonteith** [comment]
I happen to stumble on this.  I was working on a totally separate thing.  Maybe this will help.  I use copilot.microsoft.com and asked it.
` need simple code using Arduino GFX and 16 bit parallel interface for a Newhaven 2.4" 240-320 TFT display that uses a ST7789VI driver and EPS32-S3.`
You can ask it yourself or hopefully this works better.
`https://copilot.microsoft.com/?itrid=66f6087c43b44b599ba0a1a5cb82d668`

  Now, the code doesn't always work.  So you can paste error messages in the following search box and it will give you clues on how to fix.  
I chased problems for a couple of days, until I found out I needed to drop back the Espressif ESP32 Board to 2.0.17.  Which is right before the 3.x.x versions.  So much changed.  Anyway, maybe this will get you started.

 I thought of something else.  Check the manufacturers site as their docs shows the pin assignments to make sure they match the code you get from Copilot.
